metrics/release.go

```diff
@@ -17,128 +17,197 @@ limitations under the License.
+			Help:    "How long in seconds a Release takes to complete",
 			Buckets: []float64{60, 150, 300, 450, 600, 750, 900, 1050, 1200, 1800, 3600},
 		},
-		[]string{"reason", "strategy", "succeeded", "target"},
+		[]string{
+			"deployment_reason",
+			"post_actions_reason",
+			"processing_reason",
+			"release_reason",
+			"validation_reason",
+			"release_strategy",
+			"target",
+		},
 	)
 
-	ReleaseAttemptInvalidTotal = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "release_attempt_invalid_total",
-			Help: "Number of invalid releases",
+	ReleaseAttemptPostActionsExecutionDurationSeconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "release_attempt_post_actions_execution_duration_seconds",
+			Help:    "How long in seconds Release post-actions take to complete",
+			Buckets: []float64{60, 150, 300, 450, 600, 750, 900, 1050, 1200, 1800, 3600},
+		},
+		[]string{
+			"reason",
 		},
-		[]string{"reason"},
 	)
 
-	ReleaseAttemptRunningSeconds = prometheus.NewHistogram(
+	ReleaseAttemptProcessingDurationSeconds = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
-			Name:    "release_attempt_running_seconds",
-			Help:    "Release durations from the moment the release resource was created til the release is marked as running",
-			Buckets: []float64{0.5, 1, 2, 3, 4, 5, 6, 7, 10, 15, 30},
+			Name:    "release_attempt_processing_duration_seconds",
+			Help:    "How long in seconds a Release processing takes to complete",
+			Buckets: []float64{60, 150, 300, 450, 600, 750, 900, 1050, 1200, 1800, 3600},
+		},
+		[]string{
+			"reason",
+			"release_strategy",
+			"target",
 		},
 	)
 
 	ReleaseAttemptTotal = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Name: "release_attempt_total",
-			Help: "Total number of releases processed by the operator",
+			Help: "Total number of releases reconciled by the operator",
+		},
+		[]string{
+			"deployment_reason",
+			"post_actions_reason",
+			"processing_reason",
+			"release_reason",
+			"release_strategy",
+			"target",
+			"validation_reason",
 		},
-		[]string{"reason", "strategy", "succeeded", "target"},
 	)
 )
 
-// RegisterCompletedRelease decrements the 'release_attempt_concurrent_total' metric, increments `release_attempt_total`
-// and registers a new observation for 'release_attempt_duration_seconds' with the elapsed time from the moment the
-// Release attempt started (Release marked as 'Running').
-func RegisterCompletedRelease(reason, strategy, target string, startTime, completionTime *metav1.Time, succeeded bool) {
-	labels := prometheus.Labels{
-		"reason":    reason,
-		"strategy":  strategy,
-		"succeeded": strconv.FormatBool(succeeded),
-		"target":    target,
+func RegisterCompletedRelease(release *v1alpha1.Release) {
+	if !release.IsReleased() {
+		return
 	}
 
+	labels := prometheus.Labels{
+		"deployment_reason":   "",
```

Comment:
do you think instead of the empty string we should have like n/a or something that shows we are explicitly setting it to be this way, not that some operation was missed or something (not sure i am making sense). or even "success" as some of these we know are success in this function (like processing_reason)

Reply:
I think I missed a push maybe... Fixed though.